examples/demo_service/service.cpp

```diff
@@ -15,82 +15,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cmath> // rand()
+#include <sstream>
+
 #include <os>
 #include <net/inet4>
 #include <net/dhcp/dh4client.hpp>
-#include <math.h> // rand()
-#include <sstream>
-
-// An IP-stack object
-std::unique_ptr<net::Inet4<VirtioNet> > inet;
 
 using namespace std::chrono;
 
 std::string HTML_RESPONSE() {
-  int color = rand();
-  std::stringstream stream;
+  const int color = rand();
 
   /* HTML Fonts */
   std::string ubuntu_medium  = "font-family: \'Ubuntu\', sans-serif; font-weight: 500; ";
   std::string ubuntu_normal  = "font-family: \'Ubuntu\', sans-serif; font-weight: 400; ";
-  std::string ubuntu_light  = "font-family: \'Ubuntu\', sans-serif; font-weight: 300; ";
+  std::string ubuntu_light   = "font-family: \'Ubuntu\', sans-serif; font-weight: 300; ";
 
   /* HTML */
-  stream << "<html><head>"
+  std::stringstream stream;
+  stream << "<!DOCTYPE html><html><head>"
          << "<link href='https://fonts.googleapis.com/css?family=Ubuntu:500,300' rel='stylesheet' type='text/css'>"
          << "</head><body>"
-         << "<h1 style= \"color: " << "#" << std::hex << (color >> 8) << "\">"
-         <<  "<span style=\""+ubuntu_medium+"\">Include</span><span style=\""+ubuntu_light+"\">OS</span> </h1>"
+         << "<h1 style='color: #" << std::hex << (color >> 8) << "'>"
+         << "<span style='"+ubuntu_medium+"'>Include</span><span style='"+ubuntu_light+"'>OS</span></h1>"
          <<  "<h2>Now speaks TCP!</h2>"
-    // .... generate more dynamic content
-         << "<p> This is improvised http, but proper suff is in the works. </p>"
-         << "<footer><hr /> &copy; 2016, IncludeOS AS @ 60&deg; north </footer>"
-         << "</body></html>\n";
-
-  std::string html = stream.str();
-
-  std::string header="HTTP/1.1 200 OK \n "        \
-    "Date: Mon, 01 Jan 1970 00:00:01 GMT \n"      \
-    "Server: IncludeOS prototype 4.0 \n"                \
-    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT \n"   \
-    "Content-Type: text/html; charset=UTF-8 \n"     \
-    "Content-Length: "+std::to_string(html.size())+"\n"   \
-    "Accept-Ranges: bytes\n"          \
-    "Connection: close\n\n";
+         // .... generate more dynamic content
+         << "<p>This is improvised http, but proper stuff is in the works.</p>"
+         << "<footer><hr/>&copy; 2016, IncludeOS AS @ 60&deg; north</footer>"
+         << "</body></html>";
+
+  const std::string html = stream.str();
+
+  const std::string header
+  {
+    "HTTP/1.1 200 OK\n"
+    "Date: Mon, 01 Jan 1970 00:00:01 GMT\n"
+    "Server: IncludeOS prototype 4.0\n"
+    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\n"
+    "Content-Type: text/html; charset=UTF-8\n"
+    "Content-Length: "+std::to_string(html.size())+'\n'
```

Comment:
```
service.cpp:58:5: error: expected '}'
    "Accept-Ranges: bytes\n"
    ^
service.cpp:51:3: note: to match this '{'
  {
  ^
```
It looks like the compiler doesn't like that you got a char in the middle of everything. Just add a `+` after `'\n'` and it will work :)